packages/app/src/cli/commands/app/generate/schema.ts

```diff
@@ -0,0 +1,37 @@
+import {Flags} from '@oclif/core'
+import Command from '@shopify/cli-kit/node/base-command'
+import {path, error, cli} from '@shopify/cli-kit'
+import {appFlags} from '../../../flags.js'
+import {AppInterface} from '../../../models/app/app.js'
+import {load as loadApp} from '../../../models/app/loader.js'
+import {generateSchemaService} from '../../../services/generate-schema.js'
+
+export default class GenerateSchema extends Command {
+  static description = 'Generates a GraphQL schema for a function'
+
+  static flags = {
+    ...cli.globalFlags,
+    ...appFlags,
+    'api-key': Flags.string({
+      name: 'API key',
+      description: 'The API key to fetch the schema with.',
+      required: false,
+    }),
+  }
+
+  static args = [{name: 'function', required: true, description: 'The name of the function to fetch the schema for.'}]
+
+  public async run(): Promise<void> {
+    const {flags, args} = await this.parse(GenerateSchema)
+    const apiKey = flags['api-key']
+    const directory = flags.path ? path.resolve(flags.path) : process.cwd()
+    const app: AppInterface = await loadApp(directory)
+    const extension = app.extensions.function.find((f) => f.localIdentifier === args.function)
+
+    if (!extension) {
+      throw new error.Abort(`Function ${args.function} not found`)
```

Comment:
What about making this error more actionable and including the list of function IDs that are available? cc @MeredithCastile